ibis/backends/dask/execution/generic.py

```diff
@@ -0,0 +1,869 @@
+    return data.isnull()
+
+
+@execute_node.register(ops.NotNull, dd.Series)
+def execute_series_notnnull(op, data, **kwargs):
+    return data.notnull()
+
+
+@execute_node.register(ops.IsNan, (dd.Series, floating_types))
+def execute_isnan(op, data, **kwargs):
+    return np.isnan(data)
+
+
+@execute_node.register(ops.IsInf, (dd.Series, floating_types))
+def execute_isinf(op, data, **kwargs):
+    return np.isinf(data)
+
+
+@execute_node.register(ops.SelfReference, dd.DataFrame)
+def execute_node_self_reference_dataframe(op, data, **kwargs):
+    return data
+
+
+@execute_node.register(
+    ops.Contains, dd.Series, (collections.abc.Sequence, collections.abc.Set)
+)
+def execute_node_contains_series_sequence(op, data, elements, **kwargs):
+    return data.isin(elements)
+
+
+@execute_node.register(
+    ops.NotContains, dd.Series, (collections.abc.Sequence, collections.abc.Set)
+)
+def execute_node_not_contains_series_sequence(op, data, elements, **kwargs):
+    return ~(data.isin(elements))
+
+
+# Series, Series, Series
+# Series, Series, scalar
+@execute_node.register(ops.Where, dd.Series, dd.Series, dd.Series)
+@execute_node.register(ops.Where, dd.Series, dd.Series, scalar_types)
+def execute_node_where_series_series_series(op, cond, true, false, **kwargs):
+    # No need to turn false into a series, dask will broadcast it
+    return true.where(cond, other=false)
+
+
+# Series, scalar, Series
+def execute_node_where_series_scalar_scalar(op, cond, true, false, **kwargs):
+    return dd.from_array(np.repeat(true, len(cond))).where(cond, other=false)
+
+
+# Series, scalar, scalar`
+for scalar_type in scalar_types:
+    execute_node_where_series_scalar_scalar = execute_node.register(
+        ops.Where, dd.Series, scalar_type, scalar_type
+    )(execute_node_where_series_scalar_scalar)
+
+
+# scalar, Series, Series
+@execute_node.register(ops.Where, boolean_types, dd.Series, dd.Series)
+def execute_node_where_scalar_scalar_scalar(op, cond, true, false, **kwargs):
+    # Note that it is not necessary to check that true and false are also
+    # scalars. This allows users to do things like:
+    # ibis.where(even_or_odd_bool, [2, 4, 6], [1, 3, 5])
+    return true if cond else false
+
+
+# scalar, Series, scalar
+@execute_node.register(ops.Where, boolean_types, dd.Series, scalar_types)
+def execute_node_where_scalar_series_scalar(op, cond, true, false, **kwargs):
+    if cond:
+        return true
+    else:
+        # TODO double check this is the right way to do this
+        out = dd.from_array(np.repeat(false, len(true)))
+        out.index = true.index
+        return out
+
+
+# scalar, scalar, Series
+@execute_node.register(ops.Where, boolean_types, scalar_types, dd.Series)
+def execute_node_where_scalar_scalar_series(op, cond, true, false, **kwargs):
+    return dd.from_array(np.repeat(true, len(false))) if cond else false
+
+
+@execute_node.register(DaskTable, DaskClient)
+def execute_database_table_client(
```

Comment:
A bit weird to see this in the middle of logical operators. Perhaps move to another section?